### DISCUSSION TITLE
osgearth emscripten

### DISCUSSION THREAD

**Discussion by Holiflot** [pelicanmapping/osgearth#2581] [Support Q&A]
I use open scene graph and osgearth libraries to run simple.earth file in linux with this command: 
              osgearth_viewer simple.earth  
 
this works fine, however, I want to display simple.earth file in web environment by using emscripten. I installed both open scene graph and osgearth libraries using Emscripten.cmake toolchain file, however, I get this error:
 
              osgearth_viewer.js:1 Warning: GraphicsContext::WindowingSystemInterfaces::getWindowingSystemInterface() failed, no interfaces available.
              [osgEarth]* [MapNodeHelper] No earth file loaded
 
Is it possible to use open scene graph and osgearth libraries to display an .earth file in web environment using emscripten?

**Comment by gwaldron** [comment]
osgEarth does not "officially" support emscripten. But if you are able to get things working, we will be ahppy to review and integrate changes necessary to support it. Thanks!

**Comment by XenonofArcticus** [comment]
We got OSG itself working once long ago, based on an GLES2 profile (which
is effectively what WebGL, the API you'd use with Emscripten, is).

Last time we tried to get osgEarth to work with a GLES2 profile there were
some challenges. Glenn, do you know if anyone has done a GLES2 build of
recent osgEarth?

**Reply by gwaldron** [reply]
I do not.

**Reply by CaglaNurYuva** [reply]
What kind of changes has to be done in the source code to make osgearth work with GLES2 profile? If it is not too much complex, I might give it a try.

**Reply by gwaldron** [reply]
Not really sure. I think GLES2 shaders use an older standard so that would be a good chunk of the work. I think it's something you just need to start looking at to see what the challenges are.
